azure/scope/cluster.go

```diff
@@ -395,6 +396,14 @@ func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
 			NatGatewayName:    subnet.NatGateway.Name,
 			ServiceEndpoints:  subnet.ServiceEndpoints,
 		}
+		// Check if subnet is used for the private link NAT IP
+		for _, privateLink := range s.AzureCluster.Spec.NetworkSpec.APIServerLB.PrivateLinks {
+			for _, ipConfig := range privateLink.NATIPConfigurations {
+				if ipConfig.Subnet == subnetSpec.Name {
```

Comment:
Performance wise: the two outer for loops are limited, as there can be max 8 private links, and 8 NAT configs per private link. In majority of the cases it will probably be just 1 private link and 1 NAT IP config, unless there is a requirement for some very heavy traffic to the API server.

That being said, optimizing this a bit with maps would also make the code cleaner and more readable, so it could be worth it, so I'll update it.